stacks/spica/src/passport/pages/api-key-add/api-key-add.component.ts

```diff
@@ -0,0 +1,54 @@
+import {Component, OnInit, TemplateRef, ViewChild, OnDestroy} from "@angular/core";
+import {ApiKey, emptyApiKey} from "src/passport/interfaces/api-key";
+import {Router, ActivatedRoute} from "@angular/router";
+import {filter, switchMap, takeUntil} from "rxjs/operators";
+import {MockService} from "src/passport/services/api-key.service";
+import {Subject} from "rxjs";
+
+@Component({
+  selector: "app-api-key-add",
+  templateUrl: "./api-key-add.component.html",
+  styleUrls: ["./api-key-add.component.scss"]
+})
+export class ApiKeyAddComponent implements OnInit, OnDestroy {
+  @ViewChild("toolbar", {static: true}) toolbar: TemplateRef<any>;
+  public apiKey: ApiKey = emptyApiKey();
+
+  private onDestroy: Subject<void> = new Subject<void>();
+
+  constructor(
+    private router: Router,
+    private activatedRoute: ActivatedRoute,
+    private apiKeyService: MockService
+  ) {}
+
+  ngOnInit() {
+    this.getApiKeyFromId();
+  }
+
+  getApiKeyFromId() {
+    this.activatedRoute.params
+      .pipe(
+        filter(params => params.id),
+        switchMap(params => this.apiKeyService.get(params.id)),
+        takeUntil(this.onDestroy)
+      )
+      .subscribe(apiKey => {
+        this.apiKey = apiKey;
+      });
+  }
```

Comment:
There is no need to put this logic into a separate function since it is not going to be called from outside.